tests/test_model_generator.py

```diff
@@ -3722,13 +3772,277 @@ def test_modelgen_save_reload_links_03(self):
+
+        with self.assertRaisesRegex(
+            AttributeError, "'comp_opt_wprop_friend' is not set"
+        ):
+            assert tsl2.comp_opt_wprop_friend is not None  # access field
+
+    def test_modelgen_save_reload_links_05(self):
+        from models import default
+
+        # Test that backlinks become unset after save operations
+        # User.groups is a computed backlink to UserGroup.users
+
+        # Fetch Alice with her groups backlink
+        alice = self.client.get(
+            default.User.select(
+                name=True,
+                groups=True,
+            ).filter(name="Alice")
+        )
+
+        # Alice should have her groups backlink available after fetch
+        # (even if empty, it should not be None)
+        self.assertIsNotNone(alice.groups)
+
+        # Create a new UserGroup with Alice as a member
+        gr = default.UserGroup(
+            name="Dancing Banana Enthusiasts", users=[alice]
+        )
+        self.client.sync(gr)
+
+        # After saving the group, Alice's groups backlink should be unset
+        # because Alice wasn't explicitly fetched with groups after the change
+        with self.assertRaisesRegex(AttributeError, "'groups' is not set"):
+            assert alice.groups is not None  # access field
+
+        with self.assertRaisesRegex(AttributeError, "'groups' is not set"):
+            a = [u for u in gr.users if u.id == alice.id][0]
+            assert a.groups is not None  # access field
+
+        # But Alice herself should still be accessible
+        self.assertEqual(alice.name, "Alice")
+
+    def test_modelgen_save_reload_links_06(self):
+        from models import default
+
+        # Test backlink invalidation when removing a user from existing group
+        # Fetch an existing group with users and their backlinks
+
+        red = self.client.get(
+            default.UserGroup.select(
+                name=True,
+                users=lambda g: g.users.select(
+                    name=True,
+                    groups=True,
+                ),
+            ).filter(name="red")
+        )
+
+        self.assertEqual(red.name, "red")
+        self.assertEqual(
+            {u.name for u in red.users},
+            {"Alice", "Billie", "Cameron", "Dana"},
+        )
+
+        # orig_ids = {u.id for u in red.users}
+        # Find Alice in the group
+        alice = [u for u in red.users if u.name == "Alice"][0]
+        self.assertIn("red", {g.name for g in alice.groups})
+
+        # Modify and save the group
+        red.users.remove(alice)
+        self.client.sync(red)
+
+        # # After saving, Alice's groups backlink should be unset
+        # with self.assertRaisesRegex(AttributeError, "'groups' is not set"):
+        #     assert alice.groups is not None  # access field
+
+        # # But we should still have some valid data
+        # self.assertEqual(alice.name, "Alice")
+        # self.assertEqual({u.id for u in red.users}, orig_ids - {alice.id})
```

Comment:
This one is a little tricky because `alice` was definitely explicitly fetched with the backlink `groups` (not "unset"). Since it was removed from `red` it is not technically reachable from `red` anymore at the point of sync. However, I think we agreed that "reachable" should include any object being removed (since we know what they are in order to remove them). Therefore the expectation should be for the backlink of `alice` to be re-fetched.

Reply:
This one I *can* fix

Reply:
ok pushed a fix it's working now

Reply:
done, all working now